src/Swarm/TUI/View.hs

```diff
@@ -499,13 +506,18 @@ drawRecipe e inv (Recipe ins outs reqs) =
   -- If the robot doesn't have any, draw it in red.
   fmtEntityName missing ingr
     | ingr == e = withAttr deviceAttr $ txtLines nm
+    | ingr == timeE = withAttr sandAttr $ txtLines nm
     | missing = withAttr invalidFormInputAttr $ txtLines nm
     | otherwise = txtLines nm
    where
     -- Split up multi-word names, one line per word
     nm = ingr ^. entityName
     txtLines = vBox . map txt . T.words
 
+-- | Ad-hoc entity to represent time - only used in recipe drawing
+timeE :: Entity
+timeE = mkEntity (defaultEntityDisplay '.') "time piece" [] []
```

Comment:
`time piece` sounds very weird to me.  Can we just call it `ticks`?

Reply:
I still don't really like this way of drawing time as an input, but we don't need to solve it now, we can keep playing with better layouts later.